lib/Partitioner/Partitioner.cpp

```diff
@@ -948,6 +958,24 @@ llvm::Error Partitioner::Partition(CompilationContext &cctx) {
   FunctionToBackendNameMapTy funcToBackend;
   std::string origName(F_->getName().data());
   if (backends.size() == 1) {
+    if (cctx.precisionConfig.quantMode == QuantizationMode::Profile) {
+      // For quantization profiling, no need to do partitioning at this time. No
+      // matter what type of backend it is, we use CPU backend for profiling.
+      auto backendName = "CPU";
+      funcToBackend[F_] = backendName;
+      if (backendMap_.find(backendName) == backendMap_.end()) {
+        // If "CPU" backend is not in current backendMap_, add it. This is used
+        // in createDAGWithoutPartition.
+        BackendInfo backendInfo;
+        backendInfo.backend = createBackend(backendName);
+        backendMap_[backendName] = backendInfo;
```

Comment:
So you are adding logic for if we're in profiling mode in the case where we have 1 backend (here) vs. not (below). I think if we're in profiling mode we could just always set the backend to the single backend we're profiling on (e.g. CPU like here), no? Then we wouldn't need to add different cases here?

E.g. above where we call `getBackendMap()`, instead we could check if we're in profiling mode -- if not then call `getBackendMap()` as is currently done, otherwise set up backends with e.g. the CPU backend like you do, and that's all that's needed, no?

Reply:
You are right! Thanks :)